staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -174,6 +178,14 @@ func (e *watchEncoder) Encode(event watch.Event) error {
 	encodeFunc := func(obj runtime.Object, w io.Writer) error {
 		return e.doEncode(obj, event, w)
 	}
+	// if the event was transformed, then don't check if
```

Comment:
Let's be much more explicit here, I suggest switching to something like:
```
if event.Type == BOOKMARK {
  // Bookmark objects are small and we don't yet support serialization for them.
  // Additionally, we need to additionally transform them to support watch-list feature.
 transformedEvent := transformEvent(event)
 return encodeFunc(transformedEvent, e.framer)
}
```

Reply:
Does it even make sense to cache serialization for bookmarks since these events vary by RV ?

Reply:
I don't think it does, but we need to have a comment like that to remind future us about it.